lib/auth/apikey.ex

```diff
@@ -9,20 +12,19 @@ defmodule Auth.Apikey do
   schema "apikeys" do
     field :client_secret, :binary
     field :client_id, :binary
-    field :description, :string
-    field :name, :string
-    field :url, :binary
     field :person_id, :id
     field :status, :id
+    belongs_to :app, Auth.App
 
     timestamps()
   end
 
   @doc false
   def changeset(apikey, attrs) do
     apikey
-    |> cast(attrs, [:client_id, :client_secret, :name, :description, :url, :person_id])
+    |> cast(attrs, [:client_id, :client_secret, :person_id, :status])
     |> validate_required([:client_secret])
+    |> put_assoc(:app, Map.get(attrs, "app"))
```

Comment:
I think this might allow anyone to associate an API key with any "app", unless theres validation to check that the user owns the app beforehand.

It might be worth refactoring `App` and `ApiKey` out into a common context - e.g. a `Project` context with `App` and `ApiKey` models, as I feel like they are closely related with the possibility of some common functions

Reply:
@th0mas I briefly thought of removing `apikey` as a separate entity and just having `app` with a `client_id` and `client_secret` on the `app` (_i.e. only one Key pair per app_). But it was a lot of refactoring so I didn't ...

I'd actually be quite happy to re-build this whole thing from scratch in order to simplify it.
But for now I just want to finish this PR and get _something_ working. 👍

Reply:
I still think this is a potential security risk as anytime any api is key is created or changed, there is the possibility of injecting another app as its owner.